src/hotspot/share/opto/escape.cpp

```diff
@@ -800,6 +807,25 @@ void ConnectionGraph::add_final_edges(Node *n) {
       }
       break;
     }
+    case Op_Blackhole: {
+      // All blackhole arguments are globally escaping.
+      for (uint i = 0; i < n->req(); i++) {
+        Node *in = n->in(i);
+        if (in == nullptr) continue;
+
+        PointsToNode* ptn = ptnode_adr(in->_idx);
+        if (ptn == nullptr) {
+          // Not registered yet
+          add_local_var(in, PointsToNode::GlobalEscape);
+          ptn = ptnode_adr(in->_idx);
+        } else {
+          // Registered already, force the escape state
+          set_escape_state(ptn, PointsToNode::GlobalEscape NOT_PRODUCT(COMMA "blackhole"));
+        }
+        add_edge(n_ptn, ptn);
```

Comment:
Why not use `add_local_var_and_edge()` here?

Reply:
Because the input for the node might not be a `LocalVar` already, but rather `Field`, `Arraycopy`, etc.  `add_local_var_and_edge` checks this and fails on asserts. AFAICS, this is only safe to do for the node "output", but here we handle the node inputs. Maybe I should instead do what `Op_Phi` does?

Reply:
No, Phi assumes similar type of inputs. You need to do similar to `call` node in its worst case:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/escape.cpp#L1228
And you should not do `add_edge()` here since there are no data flow through Blackhole node.

Reply:
I massaged the code a bit: we only need to take care of pointer inputs to `Blackhole`, which was the issue about creating local vars before. In new code, we do the same thing as other nodes: expect the inputs to be registered already. 

Regarding `add_edge` and friends. It seems we still have to map the `Blackhole` node with `add_local_var` initially, because final processing would barf on unregistered `Blackhole` node. This forces our hand further with `add_edge`-ing its inputs, because EA further barfs when encountering the node with zero edges. This could have been avoided if we managed to process all inputs to `Blackhole` at the initial construction, but EA barfs on some of its inputs not yet registered. So we delay processing and process all inputs at final step, like AFAICS we do with call arguments. Makes sense?

Also added two IR tests.